### PR TITLE
Feat: Integrate Weave with Mirascope

### PR DESCRIPTION
**Pull request by willbakst**
- Add `with_weave` decorator for easily incorporating weave monitoring into your Mirascope workflows.
- Works with `call`, `call_async`
- Semi-works with `stream` and `stream_async` (it logs our calls, but not the model calls. this is due to how streaming works, and I'm not sure how weave handles it yet).
- Extraction is blocked by a bug with weave, but the code is ready to go once the bug is resolved.

NOTE:
This PR has the library depend on version v0.44.0 of Weave, which has a critical bug failure with pydantic.BaseModel, but the newer version hasn't been released yet. Will wait to merge/release this feature until the version with the fix is released.

### CODE REVIEW

File: mirascope/openai/calls.py [hunk 1]
```diff
@@ -68,6 +68,7 @@ def call(self, **kwargs: Any) -> OpenAICallResponse:
             OpenAIError: raises any OpenAI errors, see:
                 https://platform.openai.com/docs/guides/error-codes/api-errors
         """
+        # kwargs, tool_types = self._setup(kwargs, OpenAITool)
```
Comment (brenkao) [inline mirascope/openai/calls.py:71]: remove?

File: mirascope/weave.py
Comment (brenkao) [inline mirascope/weave.py:1]: should this be in a wandb directory?

**Review by brenkao — APPROVED** [review]

File: mirascope/openai/calls.py (hunk 1 above)
Comment (willbakst) [reply mirascope/openai/calls.py]: good catch. removed.

File: mirascope/weave.py
Comment (willbakst) [reply mirascope/weave.py:1]: done.

**Pull request closed without merging**